### ISSUE TITLE
Replace table source

### ISSUE THREAD

**Issue by jms90h5** [bokeh/bokeh#4294]
This thread https://groups.google.com/a/continuum.io/forum/#!topic/bokeh/-V-E7xzhBjg describes the requirements for updating the contents of a table.  It requires that data table is be explicitly addeded to the "current document" rather than having it done automatically the way it happens with "figure", "vplot", etc. For example:

> ```
>   curdoc().add_root(bokehTable)
> ```

Additionally replacing the source doesn't work but instead the data for the existing source needs to be replaced.  E.g.

> ```
>   bokehTable.source.data = {'AAA' : [8, 9, 10, 11], 'BBB' : [11, 21, 31, 41],'CCC' : [201,51,-31,-51]}
> ```

It would be better to have the table automatically added to the current doc and support replacing the source.

**Comment by bryevdv** [comment]
@damianavila this is a bug, and probably a one line change

**Comment by damianavila** [comment]
@bryevdv, sorry... reading too fast :wink:

**Comment by bryevdv** [comment]
@jms90h5 This seems to have been fixed, perhaps during `0.12.5`. I tested your notebook, with the updates for explicit handles:
```
handle = show(bokehTable, notebook_handle=True)
```
and later
```
push_notebook(handle=handle)
```
and the table now updates as expected. I am going to go ahead and close this but please feel free to re-open if you can still reproduce problems.

**Comment by jms90h5** [comment]
Yep.  The handle stuff seems to do the trick.
